player/csndfile.c

```diff
@@ -538,9 +538,11 @@ void csf_loop_pattern(song_t *csf, int pat, int row)
 #define SF_FAIL(name, n) \
 	({ log_appendf(4, "%s: internal error: unsupported %s %d", __FUNCTION__, name, n); return 0; })
 
-uint32_t csf_write_sample(disko_t *fp, song_sample_t *sample, uint32_t flags)
+uint32_t csf_write_sample(disko_t *fp, song_sample_t *sample, uint32_t flags, uint32_t maxlengthmask)
 {
 	uint32_t pos, len = sample->length;
+	if(maxlengthmask != UINT32_MAX)
```

Comment:
I think it would make more sense if this parameter wasn't a mask but simply an upper limit. That would also be closer to the original ModPlug source this is taken from. You could then pass `MAX(length & ~1, 0xFFFFE)` to the function for MOD.

Reply:
This parameter has only one job - limit/truncate mod samples on passed values other than default. In the future, it might be split, but till then I think that it is correct. If it is not a cardinal showstopper, I would keep it in its current form.